add waste class

TristanWebber opened the pull request:
Adding some equipment and points for waste and bins. These have been used in a RTLS (real time location system) project tracking the movement and state of bins throughout a large facility. This should be considered a starting point and is likely to be significantly refined if adapted by other use cases.

The PR has been coordinated through a working group as a colab between OAP and AIRAH.

annieDehghani commented on equip.waste.yaml:
```diff
@@ -0,0 +1,28 @@
+Bin
+ - Extend base type
+
+NAME: PWB                     ## Check for clashes
+  name: Portable Wheeled Bin
+  description: A lightweight portable bin for storing and transporting waste
```
```suggestion
BIN                
  name: Waste Bin
  description: A bin for storing and transporting waste
```
@TristanWebber a few comments
1. The main key for each YAML entry is the code.
2.  In this case, I suggest we genericize the bin object to "waste bin" rather than portable-wheeled bin to be more flexible in a wider variety of cases. Does that work for your use case?

TristanWebber replied:
Thanks - I have no issue with the generic approach.

annieDehghani commented on point.waste.yaml:
```diff
@@ -0,0 +1,66 @@
+  name: Bin fill state
+  description: Enumerated point to describe the state of the bin. Primary intent is to manage bins that may not ben in service and therefore may attract differing behaviour from analytics (e.g. 'clean', 'filling', 'full', 'dirty')
+  metadata:         ## Not really described by Haystack.
+    type:
+      - bin
+      - sensor
+      - point
```
```suggestion
    type:
      - bin
      - sensor
      - state
      - point
```
consider adding the `state` tag

annieDehghani commented:
@TristanWebber this is a great start! I made a few comments on your PR. Any new tags (such as `waste`, `weight`, `location`, etc.) will also need to be added to data/type-tags.yaml

TristanWebber commented:
Thanks Annie. Apologies it took me a while to get to this. I have accepted your changes and made others as suggested. Some notes are:

1. for `type-tags.yaml`, anything I added included the source as OAP. Is this appropriate?
2. do the attribute tags, eg `volume` and `wasteType` belong in `attr.yaml`?

TristanWebber commented:
Thanks @annieDehghani. I have now added the attributes. I renamed the `volume` attribute to `containerVolume` to reduce risk of ambiguity with the unit tag of the same name. Hopefully this allows it to be used generically as required.

Please let me know if there are any other changes required.

TristanWebber commented on data/attr.yaml:
```diff
@@ -164,3 +164,27 @@ ratedVoltage:
       - kind: Number
   unit_type: electric_potential
   
+###Common, equip-level numerical Attributes ####
+containerVolume:
+  name: Container Volume
+  description: The design volume a piece of equipment can hold. This has been added to support modelling of waste bins however
+    it has been added as a common attribute to support generic usage where needed.
+  domain: common
+  haystack:
+    type:
+     - attr
+    identification:
+      -kind: Number
+  unit_type: volume
+
+###Waste ###
+wasteType:
+  name: Waste Type
+  description: The intended waste stream within a bin or waste management process. E.g. General, Organic, Comingled, Paper & Cardboard etc
+  domain: waste
+  haystack:
+    type:
+      - attr
+    identification:
+      - kind: Str
+  unit_type: dimensionless
```
Great, I'm happy if you are!
